--- origsrc/ImageMagick-6.4.2/Magick++/Makefile.am	2008-08-16 18:02:42.000000000 -0500
+++ src/ImageMagick-6.4.2/Magick++/Makefile.am	2008-08-17 21:44:52.325109200 -0500
@@ -140,6 +140,7 @@
 	Magick++/lib/Magick++/TypeMetric.h
 
 Magick___lib_libMagick___la_LDFLAGS = \
+	-no-undefined \
 	-version-info $(MAGICK_LIBRARY_CURRENT):$(MAGICK_LIBRARY_REVISION):$(MAGICK_LIBRARY_AGE)
 Magick___lib_libMagick___la_LIBADD = $(MAGICKCORE_LIBS) $(MAGICKWAND_LIBS)
 
--- origsrc/ImageMagick-6.4.2/coders/Makefile.am	2008-07-30 21:07:16.000000000 -0500
+++ src/ImageMagick-6.4.2/coders/Makefile.am	2008-08-17 21:45:23.763212800 -0500
@@ -442,7 +442,7 @@
 coders_fpx_la_SOURCES      = coders/fpx.c
 coders_fpx_la_CPPFLAGS     = $(MODULE_EXTRA_CPPFLAGS)
 coders_fpx_la_LDFLAGS      = $(MODULECOMMONFLAGS)
-coders_fpx_la_LIBADD       = $(MAGICKCORE_LIBS) $(FPX_LIBS) -lstdc++
+coders_fpx_la_LIBADD       = $(MAGICKCORE_LIBS) $(FPX_LIBS)
 
 # GIF coder module
 coders_gif_la_SOURCES      = coders/gif.c
--- origsrc/ImageMagick-6.4.2/configure.ac	2008-08-17 10:10:46.000000000 -0500
+++ src/ImageMagick-6.4.2/configure.ac	2008-08-17 21:46:19.998667500 -0500
@@ -224,7 +224,6 @@
 cygwin_build='no'
 case "${host_os}" in
     cygwin* )
-        StaticCplusPlusLibraries='yes'
         cygwin_build='yes'
         GDI32_LIBS='-lgdi32'
     ;;
@@ -1781,7 +1780,7 @@
 GVC_PKG=""
 if test "x$with_gvc" = "xyes"; then
     AC_MSG_RESULT([-------------------------------------------------------------])
-    PKG_CHECK_MODULES(GVC,[libgvc >= 2.9.0], have_gvc=yes, have_gvc=no)
+    PKG_CHECK_MODULES(GVC,[libgvc >= 2.9.0 libagraph], have_gvc=yes, have_gvc=no)
     AC_MSG_RESULT([])
 fi
 
