--- origsrc/ImageMagick-6.4.2/Magick++/Makefile.am	2008-04-27 19:22:53.000000000 -0500
+++ src/ImageMagick-6.4.2/Magick++/Makefile.am	2008-07-16 11:51:10.187500000 -0500
@@ -139,7 +139,7 @@
 	Magick++/lib/Magick++/STL.h \
 	Magick++/lib/Magick++/TypeMetric.h
 
-Magick___lib_libMagick___la_LDFLAGS = \
+Magick___lib_libMagick___la_LDFLAGS = -no-undefined \
 	-version-info $(LIBRARY_CURRENT):$(LIBRARY_REVISION):$(LIBRARY_AGE)
 Magick___lib_libMagick___la_LIBADD = $(MAGICKCORE_LIBS) $(MAGICKWAND_LIBS)
 
--- origsrc/ImageMagick-6.4.2/coders/Makefile.am	2008-04-27 18:11:28.000000000 -0500
+++ src/ImageMagick-6.4.2/coders/Makefile.am	2008-07-16 11:50:48.718750000 -0500
@@ -442,7 +442,7 @@
 coders_fpx_la_SOURCES      = coders/fpx.c
 coders_fpx_la_CPPFLAGS     = $(MODULE_EXTRA_CPPFLAGS)
 coders_fpx_la_LDFLAGS      = $(MODULECOMMONFLAGS)
-coders_fpx_la_LIBADD       = $(MAGICKCORE_LIBS) $(FPX_LIBS) -lstdc++
+coders_fpx_la_LIBADD       = $(MAGICKCORE_LIBS) $(FPX_LIBS)
 
 # GIF coder module
 coders_gif_la_SOURCES      = coders/gif.c
--- origsrc/ImageMagick-6.4.2/configure.ac	2008-06-22 10:10:46.000000000 -0500
+++ src/ImageMagick-6.4.2/configure.ac	2008-07-16 15:20:43.765625000 -0500
@@ -12,7 +12,7 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-AC_PREREQ(2.61)
+AC_PREREQ(2.62)
 AC_INIT([ImageMagick],[6.4.2],[http://www.imagemagick.org],[ImageMagick])
 AC_CONFIG_SRCDIR([magick/MagickCore.h])
 AC_CONFIG_AUX_DIR([config])
@@ -188,7 +188,6 @@
 AM_PROG_CC_C_O
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
 AC_COMPILE_WARNINGS
 AC_PROG_LN_S
 AM_WITH_DMALLOC
@@ -210,7 +209,6 @@
 cygwin_build='no'
 case "${host_os}" in
     cygwin* )
-        StaticCplusPlusLibraries='yes'
         cygwin_build='yes'
         GDI32_LIBS='-lgdi32'
     ;;
@@ -1927,7 +1925,7 @@
 GVC_LIBS=""
 GVC_PKG=""
 if test "x$with_gvc" = "xyes"; then
-    PKG_CHECK_MODULES(GVC,[libgvc >= 2.9.0], have_gvc=yes, have_gvc=no)
+    PKG_CHECK_MODULES(GVC,[libgvc >= 2.9.0 libagraph], have_gvc=yes, have_gvc=no)
 fi
 
 if test "$have_gvc" = 'yes'; then
